### PR TITLE
Improve API ergonomics when used from Swift

### PR DESCRIPTION
**Pull request by tonyarnold**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

This PR is a stylistic update to make the exposed API a little easier to work with from Swift. This mainly involves converting the headers to vend Objective-C class properties, and annotate them for nullability.

Whereas before, the exposed API for `MSCrashes` appeared like so:

```swift
MSCrashes.setDelegate(self)
```

Now, it will show up as:

```swift
MSCrashes.delegate = self
```

Similarly for `MSCrashesDelegate`:

```swift
func crashes(_ crashes: MSCrashes!, willSend errorReport: MSErrorReport!) { … }
```

Becomes:

```
func crashes(_ crashes: MSCrashes, willSend errorReport: MSErrorReport) { … }
```

## Related PRs or issues

- This PR was borne out of the discussion in #1627.

## Misc

Please note that this is not a comprehensive set of changes, more of an exemplar of what could be done - I wanter to gather some feedback before proceeding.

### CODE REVIEW

File: AppCenterCrashes/AppCenterCrashes/MSCrashes.mm [hunk 1]
```diff
@@ -225,10 +224,22 @@ + (BOOL)hasCrashedInLastSession {
...
 
++ (void)setHasCrashedInLastSession:(BOOL)hasCrashedInLastSession {
+  [[MSCrashes sharedInstance] setDidCrashInLastSession:hasCrashedInLastSession];
+}
+
 + (BOOL)hasReceivedMemoryWarningInLastSession {
   return [[MSCrashes sharedInstance] didReceiveMemoryWarningInLastSession];
 }
 
++ (void)setHasReceivedMemoryWarningInLastSession:(BOOL)hasReceivedMemoryWarningInLastSession {
```
Comment (MatkovIvan) [inline AppCenterCrashes/AppCenterCrashes/MSCrashes.mm]: This one (and a few others) should be `readonly`. Adding setter is incorrect here

File: AppCenter/AppCenter/MSService.h [hunk 2]
```diff
@@ -11,24 +11,12 @@
...
 /**
  * Indicates whether this service is enabled.
  *
- * @return `YES` if this service is enabled, `NO` if it is not.
- *
- * @see setEnabled:
+ * The state is persisted in the device's storage across application launches.
  */
-+ (BOOL)isEnabled;
+@property(class, nonatomic, assign, setter=setEnabled:) BOOL isEnabled;
```
Comment (iamclement) [inline AppCenter/AppCenter/MSService.h]: Nit: by looking at [Apple naming conventions](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingIvarsAndTypes.html) this property should be named `enabled` with specified getter. Something like: 
```suggestion
@property(class, nonatomic, assign, getter=isEnabled:) BOOL enabled;
```
Should apply to other similar `isAdjective` properties in this PR.

Comment (tonyarnold) [reply AppCenter/AppCenter/MSService.h]: Good call. I believe this is the only `isAdjective` change in this PR.

Comment (tonyarnold) [reply AppCenter/AppCenter/MSService.h]: Sorry, the only _writable_ one.

File: AppCenterCrashes/AppCenterCrashes/MSCrashes.mm (hunk 1 above)
Comment (tonyarnold) [reply AppCenterCrashes/AppCenterCrashes/MSCrashes.mm]: Fixed by 2df5e7466e9a17a6b622a4e909666598245700dd.

File: AppCenter/AppCenter/MSService.h (hunk 2 above)
Comment (tonyarnold) [reply AppCenter/AppCenter/MSService.h]: Fixed by a9987d9625a2ea8ab1c607cd78846c692ac046e3.

**Comment by tonyarnold** [comment]
There are also a couple of properties that stick out as being in the wrong tense that it would be nice to rename, for example: 

- `hasCrashedInLastSession` → `crashedInLastSession`
- `hasReceivedMemoryWarningInLastSession` → `receivedMemoryWarningInLastSession`

But I believe that's suitable for another PR, alongside appropriate deprecation messages on the existing API.

**Comment by annakocheshkova** [comment]
Hey, thanks for the contribution. Unfortunately this PR will require changes in any application that uses our SDK which means it would be a breaking change. The build already fails at our test applications.

**Comment by tonyarnold** [comment]
No worries @annakocheshkova — it was only ever a suggestion of ways you could improve the API surface when it's exposed to Swift. Let's close this out and you folks can improve thing as/when it fits better with your schedule.

**Comment by annakocheshkova** [comment]
Hey @tonyarnold we decided to reopen it and merge to a feature branch in our repo to revisit and finish it later. Thanks for the contribution!

File: AppCenter/AppCenter/MSService.h (hunk 2 above)
Comment (iamclement) [reply AppCenter/AppCenter/MSService.h]: Sounds good thanks.

**Pull request merged**